src/main/java/com/boot/service/ItemService.java

```diff
@@ -98,35 +94,30 @@
     }
 
 
-    public ResponseEntity<?> buyItem() {
-//        UserDetailsImpl user = (UserDetailsImpl) org.springframework.security.core.context.SecurityContextHolder
-//                .getContext().getAuthentication().getPrincipal();
+    public ResponseEntity<?> buyItem(User user) {
 
-        //get information about authenticated user
-        Authentication loggedInUser = SecurityContextHolder.getContext().getAuthentication();
-        User loggedInUserFromDB = userRepository.findAllByUsername(loggedInUser.getName());
-        String userEmail = loggedInUserFromDB.getEmail();
+        String userEmail = user.getEmail();
 
         //check that the users basket not empty
-        List<Basket> userItemList = basketRepository.findItemByUserId(loggedInUserFromDB.getId());
+        List<Basket> userItemList = basketRepository.findItemByUserId(user.getId());
         if (userItemList.size() == 0) {
             return ResponseEntity.badRequest().body(new MessageResponse("The basket is empty!"));
         }
 
         //get full information about item
-        List<String> namesOfUsersItems = userItemList.stream().map(item -> item.getItem()).collect(Collectors.toList());
+        List<Item> namesOfUsersItems = userItemList.stream().map(item -> item.getItem()).collect(Collectors.toList());
         List<String> fullInformationAboutItem = new ArrayList<>();
-        for (String itemName : namesOfUsersItems) {
-            fullInformationAboutItem.add(itemRepository.findAllByName(itemName).getName());
-            fullInformationAboutItem.add(itemRepository.findAllByName(itemName).getDescription());
+        for (Item usersItem : namesOfUsersItems) {
+            fullInformationAboutItem.add(itemRepository.findAllByName(usersItem.getName()).getName());
+            fullInformationAboutItem.add(itemRepository.findAllByName(usersItem.getName()).getDescription());
         }
 
         // send inf about item on the users email
         String message = "the products you bought: " + fullInformationAboutItem;
         // fullInformationAboutItem.stream().forEach(item -> System.out.println(item));
         mailSender.send(userEmail, "your list of purchased items ", message);
         //clean the basket
```

Comment:
useless comment. basketRepository.deleteAllByUserId(user.getId()); - from method it is obvious, that you delete baskets by user.  No need comments here